src/DataLayouts/cuda.jl

```diff
@@ -37,42 +45,32 @@ Base.similar(
     dims::Dims{N},
 ) where {T, N, B} = similar(CUDA.CuArray{T, N, B}, dims)
 
-function knl_copyto!(dest, src)
-
-    #=
-    nij, nh = size(dest)
-
-    thread_idx = CUDA.threadIdx().x
-    block_idx = CUDA.blockIdx().x
-    block_dim = CUDA.blockDim().x
-
-    # mapping to global idx to make insensitive
-    # to number of blocks / threads per device
-    global_idx = thread_idx + (block_idx - 1) * block_dim
-
-    nx, ny = nij, nij
-    i = global_idx % nx == 0 ? nx : global_idx % nx
-    j = cld(global_idx, nx)
-    h = ((global_idx-1) % (nx*nx)) + 1
-    =#
-
-    i = CUDA.threadIdx().x
-    j = CUDA.threadIdx().y
-
-    h = CUDA.blockIdx().x
-
-    p_dest = slab(dest, h)
-    p_src = slab(src, h)
-
-    @inbounds p_dest[i, j] = p_src[i, j]
+function knl_copyto!(pdest, psrc)
+    gidx = threadIdx().x + (blockIdx().x - 1) * blockDim().x
+    if gidx < length(pdest)
+        @inbounds pdest[gidx] = psrc[gidx]
+    end
     return nothing
 end
 
 function Base.copyto!(
     dest::IJFH{S, Nij},
     bc::Union{IJFH{S, Nij, A}, Base.Broadcast.Broadcasted{IJFHStyle{Nij, A}}},
 ) where {S, Nij, A <: CUDA.CuArray}
-    _, _, _, _, Nh = size(bc)
-    CUDA.@cuda threads = (Nij, Nij) blocks = (Nh,) knl_copyto!(dest, bc)
+    pdest, pbc = parent(dest), parent(bc)
```

Comment:
this won't work for `Broadcasted` objects.

Reply:
Yeah, this is breaking..